validation/validate_dc2_run2.2i_object_table.py

```diff
@@ -0,0 +1,870 @@
+        ax.hist(
+            [good[field], stars[field], galaxies[field]],
+            label=labels,
+            linestyle=ls,
+            **hist_kwargs,
+        )
+
+    ax.set_ylim(100, ax.get_ylim()[1])
+
+    ax.set_xlabel(f"{filt}-band Moments: Ixx, Iyy, Ixy [pixels^2]")
+    ax.set_ylabel("objects / bin")
+    if legend:
+        ax.legend()
+
+    plt.tight_layout()
+    if plotname is not None:
+        plt.savefig(plotname)
+        plt.clf()
+
+
+def plot_psf_fwhm(
+    good,
+    filters,
+    colors=("purple", "blue", "green", "orange", "red", "brown"),
+    plotname=None,
+):
+    for filt, color in zip(filters, colors):
+        psf_fwhm = np.array(good[f"psf_fwhm_{filt}"])
+        (w,) = np.where(np.isfinite(psf_fwhm) & (psf_fwhm < 3))
+        sns.distplot(psf_fwhm[w], label=filt, color=color)
+    plt.xlabel("PSF FWHM [arcsec]")
+    plt.ylabel("normalized object density")
+    plt.legend()
+
+    plt.tight_layout()
+    if plotname is not None:
+        plt.savefig(plotname)
+        plt.clf()
+
+
+def plot_ellipticity_filters(good, stars, galaxies, filters, plotname=None):
+    fig, axes = plt.subplots(2, 3, figsize=(12, 6))
+    legend = True
+    for ax, filt in zip(axes.flat, filters):
+        plot_ellipticity(good, stars, galaxies, filt, ax=ax, legend=legend)
+        legend = False
+    plt.tight_layout()
+
+    if plotname is not None:
+        plt.savefig(plotname)
+        plt.clf()
+
+
+def plot_gmr_hist(stars, galaxies, plotname=None):
+    plt.hist(
+        [galaxies["mag_g"] - galaxies["mag_r"], stars["mag_g"] - stars["mag_r"]],
+        label=["galaxies", "stars"],
+        histtype="step",
+        bins=np.linspace(-5, +5, 51),
+    )
+    plt.xlabel("g-r")
+    plt.ylabel("objects / bin")
+    if plotname is not None:
+        plt.savefig(plotname)
+        plt.clf()
+
+
+def plot_gmr_cmodel(stars, figsize=(6, 6), plotname=None):
+    plt.figure(figsize=figsize)
+    plt.hexbin(
+        stars["mag_g"] - stars["mag_r"],
+        stars["mag_i"] - stars["mag_i_cModel"],
+        extent=(-2, +3, -0.5, +5),
+        bins="log",
+    )
+    plt.xlabel("g-r")
+    plt.ylabel("mag_i[_psf] - mag_i_CModel")
+    plt.colorbar(label="objects / bin")
+
+    if plotname is not None:
+        plt.savefig(plotname)
+        plt.clf()
+
+
+def plot_shape_filters(good, stars, galaxies, filters, plotname=None):
+    # ## Shape Parameters
+    #
+    # Ixx, Iyy, Ixy
+    fig, axes = plt.subplots(2, 3, figsize=(12, 6))
+    legend = True
+    for ax, filt in zip(axes.flat, filters):
+        plot_shape(good, stars, galaxies, filt, ax=ax, legend=legend)
+        legend = False
+
+    if plotname is not None:
+        plt.savefig(plotname)
+        plt.clf()
+
+
+def run():
+    suffix = "pdf"
+    # Processing the first 78 tracts from Run 2.2i DR6: "DR6a"
+    data_release = "DC2_Run2.2i_DR6a"
+    sampling_factor = 1
+
+    filters, df = load_data(sampling_factor=sampling_factor)
```

Comment:
Here one should pass the `data_release` variable to `load_data` otherwise it leaves impression that `load_data` can magically know which data release to use.